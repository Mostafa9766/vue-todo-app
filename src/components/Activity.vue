<template>
    <div class="main">
        <h1>ToDo APP</h1>
        <div v-for="item in activities">
            <el-row style="margin-bottom: 0.51rem;" :class="getClass(item)">
                <el-col :span="6">
                    {{item.name}}
                </el-col>
                <el-col :span="6">
                    {{item.date}}
                </el-col>
                <el-col :span="6">
                    <el-button type="danger" icon="el-icon-delete" @click="removeActivity(item)" circle
                               size="small"></el-button>
                </el-col>
                <el-col :span="6">
                    <el-button type="success" icon="el-icon-check" circle @click="completeActivity(item)"
                               size="small"></el-button>
                </el-col>
            </el-row>
        </div>

        <div>
            <el-row style="width: 60%; margin-left: 40%; margin-bottom: 0.24rem;" >
                <el-col :span="3">
                    Completed :
                </el-col>
                <el-col :span="2">
                    {{completed}}
                </el-col>
                <el-col :span="3">
                    Total :
                </el-col>
                <el-col :span="2">
                    {{total}}
                </el-col>
            </el-row>
        </div>

        <div v-if="isWrongActivity" style="margin-bottom: 0.2rem; width: 80%; margin-left: 10%; text-align: center">
            <el-alert
                    :title="errorMessage"
                    type="error"
                    :center="true"
                    show-icon
                    :closable="false">
            </el-alert>
        </div>

        <div>
            <el-row>
                <el-col :span="3">
                    Activity:
                </el-col>

                <el-col :span="8">
                    <el-input placeholder="Please input the activity"
                              v-model="activity" size="mini">
                    </el-input>
                </el-col>

                <el-col :span="3">
                    Date:
                </el-col>

                <el-col :span="8">
                    <el-input placeholder="Please input the date: dd/mm/yyyy"
                              v-model="date" size="mini">
                    </el-input>
                </el-col>
            </el-row>

            <el-button type="primary" icon="el-icon-circle-plus-outline" circle
                       @click="addActivityMethod()"></el-button>
        </div>

    </div>
</template>

<script>
    import {mapGetters, mapActions} from 'vuex'

    export default {
        name: 'Activity',
        data() {
            return {
                activity: '',
                wrong: false,
                date: '',
                errorMessage: ''
            }
        },
        computed: {
            ...mapGetters({
                'activities': 'getActivities'
            }),
            isWrongActivity() {
                return this.wrong
            },
            completed() {
                return this.activities.filter((val) => val.completed === true).length;
            },
            total() {
                return this.activities.length;
            }


        },
        methods: {
            ...mapActions(['addActivity', 'deleteActivity', 'changeActivityState']),
            addActivityMethod() {
                if (this.validateData() === true) {
                    const activity = {
                        name: this.activity,
                        completed: false,
                        date: this.date
                    };

                    this.addActivity({activity});
                    this.activity = '';
                    this.date= '';
                    this.wrong = false;

                } else {
                    this.wrong = true;
                    this.setMessageError();
                }
            },

            removeActivity(item) {
                this.deleteActivity({activity: item})
            },

            completeActivity(item) {
                this.changeActivityState({activity: item})
            },

            getClass(item) {
                return item.completed === true ? 'completed' : ''
            },

            validateData () {
                if (this.activity !== '' && this.date !== '') {
                    return true;
                } else {
                    return false;
                }
            },

            setMessageError() {
                if (this.activity === '' && this.date === '') {
                    this.errorMessage= 'The activity && the date are empty';
                } else {
                    if (this.activity === '') {
                        this.errorMessage = 'The activity is empty';
                    } else {
                        this.errorMessage = 'The date is empty';

                    }
                }
            }

        }
    }
</script>

<style>
    .main {
        text-align: center;
    }

    .completed {
        background: rgba(0, 0, 0, 0.3);
    }

</style>