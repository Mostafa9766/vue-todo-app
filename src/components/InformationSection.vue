<template>
  <div>
    <div>
      <el-row type="flex" justify="center" class="counterSection">
        <el-col :span="4">
          Completed :
        </el-col>
        <el-col :span="2">
          {{completed}}
        </el-col>
        <el-col :span="3">
          Total :
        </el-col>
        <el-col :span="2">
          {{total}}
        </el-col>
      </el-row>
    </div>
  </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
  name: 'InformationSection',
  computed: {
    ...mapState(['activities']),
    completed() {
      return this.activities.filter((val) => val.completed === true).length;
    },
    total() {
      return this.activities.length;
    },
  },
};
</script>

<style>

.counterSection {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
}
</style>
