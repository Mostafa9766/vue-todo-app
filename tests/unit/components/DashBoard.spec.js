import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import ElementUi from 'element-ui';

import DashBoard from '@/components/DashboardComponent.vue';
import Activity from '@/components/ActivityComponent.vue';
import InformationSection from '@/components/InformationSection.vue';
import InputSection from '@/components/InputSection.vue';

describe('DashBoard.vue', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);
  localVue.use(ElementUi);

  let store;
  let state;

  beforeEach(() => {
    state = {
      activities: () => [],
      errorMessage: () => '',
    };

    store = new Vuex.Store({
      state,
    });
  });

  it('render DashBoard component', () => {
    const wrapper = shallowMount(DashBoard, {
      localVue,
      store,
      components: {
        Activity,
        InformationSection,
        InputSection,
      },
    });

    expect(wrapper).toBeTruthy();
  });
});
