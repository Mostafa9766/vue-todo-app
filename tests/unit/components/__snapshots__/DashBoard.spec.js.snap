// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashBoard.vue render DashBoard component 1`] = `
<div class="main container">
  <h1>ToDo APP</h1>
  <div>
    <el-alert-stub title="There are not activities at this moment" description="" type="info" closetext="" showicon="true" center="true" effect="light"></el-alert-stub>
  </div>
  <!---->
  <!---->
  <information-section-stub></information-section-stub>
  <input-section-stub></input-section-stub>
</div>
`;
